code/modules/organs/ailments/_ailment.dm

```diff
@@ -0,0 +1,96 @@
+/datum/ailment
+	var/name
+	var/timer_id
+	var/min_time = 2 MINUTES
+	var/max_time = 5 MINUTES
+	var/category = /datum/ailment
+	var/obj/item/organ/organ
+
+	// Requirements before applying to a target.
+	var/list/applies_to_organ
+	var/specific_organ_subtype = /obj/item/organ/external
+	var/affects_robotics = FALSE
+
+	var/treated_by_item_type
+	var/treated_by_item_cost = 1
+	var/treated_by_reagent_type
+	var/treated_by_reagent_dosage = 1
+	var/initial_ailment_message = "Your $ORGAN$ doesn't feel quite right..."
+	var/third_person_treatement_message = "$USER$ treats $TARGET$'s ailment with $ITEM$."
+	var/self_treatement_message = "$USER$ treats $USER_HIS$ ailment with $ITEM$."
+	var/medication_treatment_message = "Your ailment abates."
+
+/datum/ailment/New(var/obj/item/organ/_organ)
+	..()
+	if(_organ)
+		organ = _organ
+		if(organ.owner)
+			to_chat(organ.owner, SPAN_WARNING(capitalize(replacetext(initial_ailment_message, "$ORGAN$", organ.name))))
+			begin_ailment_event()
+
+/datum/ailment/proc/can_apply_to(var/obj/item/organ/_organ)
+	if(specific_organ_subtype && !istype(_organ, specific_organ_subtype))
+		return FALSE
+	if(affects_robotics && !BP_IS_PROSTHETIC(organ))
+		return FALSE
+	if(!affects_robotics && BP_IS_PROSTHETIC(organ))
+		return FALSE
+	if(length(applies_to_organ) && !(_organ?.organ_tag in applies_to_organ))
+		return FALSE
+	return TRUE
+
+/datum/ailment/Destroy(force)
+	. = ..()
+	if(organ)
+		LAZYREMOVE(organ.ailments, src)
+		organ = null
+	deltimer(timer_id)
+	timer_id = null
```

Comment:
Aren't these cleaned automatically? Even if not, all this should be above the `..()`.

Reply:
I don't know, this code is copied over from the original implementation.

Reply:
They're not cleaned per-se, but nulling it out is more or less pointless - this is just a number that shouldn't be reused within a reasonable timespan.